browser/brave_content_browser_client.cc

```diff
@@ -1136,6 +1131,12 @@ BraveContentBrowserClient::CreateThrottlesForNavigation(
     throttles.push_back(std::move(debounce_throttle));
   }
 
+  // Rewards
+  if (auto rewards_protocol_throttle = brave_rewards::
+          RewardsProtocolNavigationThrottle::MaybeCreateThrottleFor(handle)) {
+    throttles.insert(throttles.begin(), std::move(rewards_protocol_throttle));
```

Comment:
Curious: why not `push/emplace_back`?

Reply:
Can't we do just:
```cpp
throttles.emplace_back(std::make_unique<RewardsProtocolNavigationThrottle>(handle));
```

Reply:
Okay, then I think it can just be:
```cpp
throttles.insert(throttles.begin(), std::make_unique<RewardsProtocolNavigationThrottle>(handle));
```

Reply:
@szilardszaloki shouldn't we check for throttle first ? if it's null or not

Reply:
No, you'd pass the result of `std::make_unique<>()` to `std::vector<T>::insert()` directly.

Reply:
in that case, we don't need `MaybeCreateThrottleFor` at all.

Reply:
Exactly.